californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java

```diff
@@ -213,7 +213,7 @@ public void run() {
 	public void receiveRequest(final Exchange exchange, final Request request) {
 
 		if (request.isDuplicate()) {
-			if (exchange.getSendNanoTimestamp() > request.getReceiveNanoTimestamp()) {
+			if (exchange.getSendNanoTimestamp() > request.getNanoTimestamp()) {
```

Comment:
Should we do that  for response too  ? I'm not sure

Reply:
Maybe. I try it.